services/scabbard/libscabbard/src/service/factory.rs

```diff
@@ -156,15 +156,25 @@ impl ServiceArgValidator for ScabbardArgValidator {
         let admin_keys_str = args
             .get("admin_keys")
             .ok_or_else(|| ServiceArgValidationError("admin_keys argument not provided".into()))?;
-        let admin_keys: Vec<String> = serde_json::from_str(admin_keys_str).map_err(|err| {
-            ServiceArgValidationError(format!("failed to parse admin_keys list: {}", err,))
-        })?;
+
+        let admin_keys = {
+            if admin_keys_str.starts_with('[') {
```

Comment:
Can we pull this list parsing out to a function? (you do the same operation three times here)

Reply:
added